pcdsdevices/pulsepicker.py

```diff
@@ -0,0 +1,155 @@
+import logging
+
+from ophyd.device import Component as Cmp, FormattedComponent as FCmp
+from ophyd.signal import EpicsSignal, EpicsSignalRO
+from ophyd.status import SubscriptionStatus, wait as status_wait
+
+from .inout import InOutRecordPositioner, InOutPVStatePositioner
+
+logger = logging.getLogger(__name__)
+
+
+class PulsePicker(InOutPVStatePositioner):
+    """
+    Device that can open/close in response to event codes to let certain pulses
+    through and block others.
+    """
+    blade = Cmp(EpicsSignalRO, ':READ_DF')
+    mode = Cmp(EpicsSignalRO, ':SD_SIMPLE')
+
+    cmd_reset = Cmp(EpicsSignal, ':RESET_PG')
+    cmd_open = Cmp(EpicsSignal, ':S_OPEN')
+    cmd_close = Cmp(EpicsSignal, ':S_CLOSE')
+    cmd_flipflop = Cmp(EpicsSignal, ':RUN_FLIPFLOP')
+    cmd_burst = Cmp(EpicsSignal, ':RUN_BURSTMODE')
+    cmd_follower = Cmp(EpicsSignal, ':RUN_FOLLOWERMODE')
+
+    states_list = ['OPEN', 'CLOSED']
+    in_states = ['CLOSED']
+    out_states = ['OPEN']
+    _states_alias = {'CLOSED': ['CLOSED', 'IN']}
+    _state_logic = {'blade': {0: 'OPEN',
+                              1: 'CLOSED',
+                              2: 'CLOSED'}}
+
```

Comment:
Is `blade` a read_attr by default?

Reply:
I've only kept the `_default_read_attrs = ['state']` from the parent classes.